File: client/clojure/src/cdisc_library_api/specs/export_sdtm_class_variables_row.clj
(ns cdisc-library-api.specs.export-sdtm-class-variables-row
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def export-sdtm-class-variables-row-data
  {
   (ds/opt :Version) string?
   (ds/opt :VariableOrder) string?
   (ds/opt :Class) string?
   (ds/opt :DatasetName) string?
   (ds/opt :VariableName) string?
   (ds/opt :VariableLabel) string?
   (ds/opt :Type) string?
   (ds/opt :ControlledTermsCodelistorFormat) string?
   (ds/opt :Role) string?
   (ds/opt :RoleDescription) string?
   (ds/opt :Description) string?
   })

(def export-sdtm-class-variables-row-spec
  (ds/spec
    {:name ::export-sdtm-class-variables-row
     :spec export-sdtm-class-variables-row-data}))

File: client/scalaz/src/main/scala/org/openapitools/client/api/ExportSdtmClassVariablesRow.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ExportSdtmClassVariablesRow._

case class ExportSdtmClassVariablesRow (
  version: Option[String],
variableOrder: Option[String],
`class`: Option[String],
datasetName: Option[String],
variableName: Option[String],
variableLabel: Option[String],
`type`: Option[String],
controlledTermsCodelistOrFormat: Option[String],
role: Option[String],
roleDescription: Option[String],
description: Option[String])

object ExportSdtmClassVariablesRow {
  import DateTimeCodecs._

  implicit val ExportSdtmClassVariablesRowCodecJson: CodecJson[ExportSdtmClassVariablesRow] = CodecJson.derive[ExportSdtmClassVariablesRow]
  implicit val ExportSdtmClassVariablesRowDecoder: EntityDecoder[ExportSdtmClassVariablesRow] = jsonOf[ExportSdtmClassVariablesRow]
  implicit val ExportSdtmClassVariablesRowEncoder: EntityEncoder[ExportSdtmClassVariablesRow] = jsonEncoderOf[ExportSdtmClassVariablesRow]
}
